src/WebJobs.Extensions.OpenAI/Models/ChatMessageTableEntity.cs

```diff
@@ -41,4 +41,9 @@ class ChatMessageTableEntity : ITableEntity
     /// </summary>
     public ETag ETag { get; set; }
 
+    // <summary>
+    /// Gets when chat message was created.
+    /// </summary>
+    public DateTime CreatedAt { get; set; }
```

Comment:
How is this different from the `Timestamp` property?

Reply:
I realized we don't need that anymore. Thanks for catching it!

Reply:
just kidding so we actually do need it. Otherwise the build randomly fails in the E2E test since the test is dependent on the `lastUpdated` field of the state to try to get the unread messages. The timestamp that is on each table entity is the timestamp for when the batch transaction happens, but not the actual time for when the message is created. This would cause the E2E test to be flakey since sometimes the timestamp manages to successfully filter the entities.

 So for when we get the state dependent on the timestamp we use the `CreatedAt` field instead of the `timestamp` field to see if we want to use that message. Does that make sense?

Reply:
Makes sense - thanks for confirming!